internal/operator-controller/features/features.go

```diff
@@ -36,3 +39,20 @@ var OperatorControllerFeatureGate featuregate.MutableFeatureGate = featuregate.N
 func init() {
 	utilruntime.Must(OperatorControllerFeatureGate.Add(operatorControllerFeatureGates))
 }
+
+// LogFeatureGateStates logs the state of all known feature gates.
+func LogFeatureGateStates(log logr.Logger, fg featuregate.FeatureGate) {
+	// Sort the keys for consistent logging order
+	featureKeys := make([]featuregate.Feature, 0, len(operatorControllerFeatureGates))
+	for k := range operatorControllerFeatureGates {
+		featureKeys = append(featureKeys, k)
+	}
+	sort.Slice(featureKeys, func(i, j int) bool {
+		return string(featureKeys[i]) < string(featureKeys[j]) // Sort by string representation
+	})
+
+	log.Info("Feature Gates Status:")
+	for _, feature := range featureKeys {
+		log.Info("  ", "feature", string(feature), "enabled", fg.Enabled(feature))
+	}
```

Comment:
This is not how structured loggers should be used.

It should be something more like this:
```go
	var featurePairs []string
	for _, feature := range featureKeys {
		featurePairs = append(featurePairs, string(feature), fmt.Sprintf("%v", fg.Enabled(feature)))
	}
	log.Info("feature gate status", featurePairs...)
```

Reply:
Changed